### ISSUE TITLE
Scripts for Deployment Process

### ISSUE THREAD

**Issue by mattiasaxell** [stefanw/froide#162]
@stefanw @fin we've me & @robinharms have been e-mailing about issues on how to get FROIDE up and running. 

Could you please make MTA config, Ansible Script accessible or share with us?

@robinharms please add any extra script requirements that would help you make it work.

**Comment by robinharms** [comment]
Just a list of dependencies (like libmagic) and needed services. Example configs would be really useful.

**Comment by stefanw** [comment]
Currently FragDenStaat.de deployment is still a fab script, but it only takes care of deployment (= updating code, dependencies, running migrations etc. and restarting services), not of provisioning (which is the hard part.

I just updated [froide-theme](https://github.com/okfde/froide-theme) and pushed my current (not yet fully working) version of a [dockerisation of froide and fragdenstaat](https://github.com/okfde/fragdenstaat.de-infra). This should contain some hints on how to set it up.

I'd be happy to continue to develop this with you guys, if it makes sense to you. Ansible may be another good way to get things started.

**Comment by ryankanno** [comment]
Aloha! :D We're thinking about deploying Froide in Hawaii. Awesome work, btw.  I have a set of (fairly opinionated) Ansible scripts that currently let us deploy this - http://beta.uipa.org using Terraform  + Ansible to Digital Ocean. They may have to be tweaked a little, but let me know the best way to share them.

**Comment by fin** [comment]
@ryankanno do you think you could share it as a gist?
i'm in the process of looking over my scripts to make them publish-able, but i need to be sure there are no keyfiles in there, etc.

**Comment by ryankanno** [comment]
@fin Absolutely. I'm in the process of cleaning them up right now. It'll be a Vagrantfile with a bunch of Ansible tasks. I'll link the gist here once I get everything cleaned up. I also have a topic branch that I'll link to see the commits, etc. What's the best strategy to update the Froide settings? Update settings.py with a new class re: class Vagrant(Base):

**Comment by fin** [comment]
@ryankanno re froide settings: i'd suggest creating a repo based on froide-theme and adding custom config there, as i did here: https://github.com/fin/fragdenstaat.at/blob/master/froide_theme/settings.py
(and as stefan does with fragdenstaat.de as well)

**Comment by ryankanno** [comment]
@fin Thanks for that tip. Definitely going down that route. There were 22 files changed so I wasn't sure how that would look as a gist. I did check all my changes into a [branch](https://github.com/ryankanno/froide/commit/5595e5d214551cd1f1ad0f03d91c0a3394d5e925). I tried to document what I did in a Vagrantfile [readme](https://github.com/ryankanno/froide/blob/vagrant/Vagrantfile.README.md).

**Comment by ryankanno** [comment]
@fin As I was going through setting up the application and what not, I also realized that we didn't set up elasticsearch. Will be updating the branch to take care of that.

**Comment by fin** [comment]
@robinharms @mattiasaxell i'll be (and have been) hard to reach for a week or so due to the death of a dear friend. sorry for the delay.
i was 95% done with the deploy script, but it seems like @ryankanno has you mostly covered.
(my deployment even runs with whoosh instead of elasticsearch - which should currently be the default - without performance issues for now)

**Comment by mattiasaxell** [comment]
I'm sorry for your loss @fin, sad to hear that! Thanks a lot for taking time to answer.
@ryankanno is your custom config working properly in production stage? In that case we'd love to have a look over it. Is it a little cleaned up now and ready for release?

**Comment by ryankanno** [comment]
@fin My condolences to you and yours. 

@mattiasaxell @robinharms I haven't had time to configure the search index yet in this [branch](https://github.com/ryankanno/froide/commit/5595e5d214551cd1f1ad0f03d91c0a3394d5e925). I'll be getting to it shortly. With that said, I've set it up with Vagrant in that branch, so you should be able to play around / customize it in isolation fairly easily. Let me know if you need anything else from me.

**Comment by mattiasaxell** [comment]
@fin beautiful gift at the online memorial website!

@ryankanno alright, so it would be soon ready to be up and running in production mode? We would love to have your list of dependencies, is that already set up in the branch? To me it looks like it, but I'm new to this area. Also, would it be much work to change it to Debian? :)

**Comment by ryankanno** [comment]
@mattiasaxell If you want to ping me off this github issue, I can try to help you get it setup.  I'm in the process myself of trying to get it productionized. As a note, that Vagrantfile doesn't deal with securing the machine since that's a bit out of scope (and a lot of personal idealogy). :)

**Comment by mattiasaxell** [comment]
@ryankanno I'd love to, how can I ping you outside the issue? I'm thiking maybe it is wise if you @robinharms if you can help. He will be looking at your script when he tries to set it up for production purpose on Tue 6th via Debian-based VPS. Do you think you can have it ready by then?

Securing the machine, do you mean as in firewall or stability? I'm new to system deployments. :)
